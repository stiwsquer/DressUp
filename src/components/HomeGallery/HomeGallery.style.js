import styled from "styled-components";
import { flex } from "../../GlobalStyles";

export const HomeGallerySection = styled.section`
  display: grid;
  min-height: 100vh;
  grid-template-columns: repeat(3, 1fr);
  width: 95%;
  margin: auto;

  p {
    margin: 0rem 6rem;
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
  .gallery1 {
    grid-column: 1/2;
    grid-row: 1/3;
  }
  .gallery2 {
    grid-column: 2/3;
    grid-row: 2/3;
  }
  .gallery3 {
    grid-column: 3/4;
    grid-row: 1/2;
  }
  .gallery-new-shoes {
    grid-column: 2/3;
    grid-row: 1/2;
  }

  .gallery-accessories {
    grid-column: 3/4;
    grid-row: 2/3;
  }

  .gallery-accessories,
  .gallery-new-shoes {
    ${flex({ direction: "column", justify: "center", align: "center" })}
    text-align: center;
    font-size: 1.5rem;
  }
  .gallery-accessories h2,
  .gallery-new-shoes h2 {
    margin: 2rem;
    letter-spacing: 0.2rem;
  }

  button {
    margin-top: 2rem;
  }

  @media screen and (max-width: 900px) {
    img,
    div {
      grid-row: auto;
      grid-column: auto;
    }
    p {
      margin: 1rem;
    }
    .gallery1 {
      display: none;
    }
    grid-template-columns: repeat(auto-fit, minmax(30rem, 1fr));
  }
`;
