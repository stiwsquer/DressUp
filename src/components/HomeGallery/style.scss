@import "../../styles/global.scss";
.home-gallery {
  display: grid;
  // gap: 3rem;
  min-height: 100vh;
  grid-template-columns: repeat(3, 1fr);
  width: 95%;
  margin: auto;

  p {
    margin: 0rem 6rem;
  }
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
  .gallery1 {
    grid-column: 1/2;
    grid-row: 1/3;
  }
  .gallery2 {
    grid-column: 2/3;
    grid-row: 2/3;
  }
  .gallery3 {
    grid-column: 3/4;
    grid-row: 1/2;
  }
  .gallery-new-shoes {
    grid-column: 2/3;
    grid-row: 1/2;
  }

  .gallery-accessories {
    grid-column: 3/4;
    grid-row: 2/3;
  }

  .gallery-accessories,
  .gallery-new-shoes {
    @include flex(column, center, center);
    text-align: center;
    font-size: 1.5rem;
    h2 {
      margin: 2rem;
      letter-spacing: 0.2rem;
    }
  }
  button {
    margin-top: 2rem;
  }
}

@media screen and (max-width: 900px) {
  .home-gallery {
    // width: 100vw;
    img,
    div {
      grid-row: auto;
      grid-column: auto;
    }

    p {
      margin: 1rem;
    }

    .gallery1 {
      display: none;
    }

    grid-template-columns: repeat(auto-fit, minmax(30rem, 1fr));
  }
}
