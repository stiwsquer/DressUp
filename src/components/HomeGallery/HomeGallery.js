import React from "react";
import Button from "../Button/Button";
import "./style.scss";

export default React.memo(function HomeGallery() {
  return (
    <section className="home-gallery">
      <div className="gallery-new-shoes">
        <h2>NEW SHOES</h2>
        <p>
          We just dropped new fall shoe styles, and they are all ON SALE! Shop
          new combat boots, booties, mules, sneakers & more!
        </p>
        <Button>SHOP SHOES</Button>
      </div>
      <div className="gallery-accessories">
        <h2>Accessories</h2>
        <p>
          Take your fall outfit to the next level with new accessories! The
          perfect trendy handbag, dainty gold necklace, coin bracelet or
          statement earrings will upgrade any look.
        </p>
        <Button>SHOP ACCESSORIES</Button>
      </div>
      <img className="gallery1" src="./img/home/gallery/gallery-1.2.jpg" />
      <img className="gallery2" src="./img/home/gallery/gallery-2.1.jpg" />
      <img className="gallery3" src="./img/home/gallery/gallery-3.1.jpg" />
    </section>
  );
});
