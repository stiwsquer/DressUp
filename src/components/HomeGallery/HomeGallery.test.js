import HomeGallery from "./HomeGallery";
import { BrowserRouter as Router } from "react-router-dom";

it("renders HomeGallery component without crashing", () => {
  shallow(
    <Router>
      <HomeGallery />
    </Router>
  );
});

it("contains two Button components", () => {
  const wrapper = mount(
    <Router>
      <HomeGallery />
    </Router>
  );
  expect(wrapper.find("Button")).toHaveLength(2);
});

it("contains paragraph", () => {
  const wrapper = mount(
    <Router>
      <HomeGallery />
    </Router>
  );
  expect(
    wrapper.contains(
      <p>
        We just dropped new fall shoe styles, and they are all ON SALE! Shop new
        combat boots, booties, mules, sneakers & more!
      </p>
    )
  ).toBe(true);
});
 
it("HomeGallery snapshop - shallow", () => {
  const wrapper = shallow(
    <Router>
      <HomeGallery />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("HomeGallery snapshop - render", () => {
  const wrapper = render(
    <Router>
      <HomeGallery />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});

it("HomeGallery snapshop - mount", () => {
  const wrapper = mount(
    <Router>
      <HomeGallery />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
