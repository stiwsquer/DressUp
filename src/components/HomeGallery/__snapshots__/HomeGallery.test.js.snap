// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeGallery snapshop - mount 1`] = `
<section
  className="home-gallery"
>
  <div
    className="gallery-new-shoes"
  >
    <h2>
      NEW SHOES
    </h2>
    <p>
      We just dropped new fall shoe styles, and they are all ON SALE! Shop new combat boots, booties, mules, sneakers & more!
    </p>
    <button>
      SHOP SHOES
    </button>
  </div>
  <div
    className="gallery-accessories"
  >
    <h2>
      Accessories
    </h2>
    <p>
      Take your fall outfit to the next level with new accessories! The perfect trendy handbag, dainty gold necklace, coin bracelet or statement earrings will upgrade any look.
    </p>
    <button>
      SHOP ACCESSORIES
    </button>
  </div>
  <img
    className="gallery1"
    src="./img/home/gallery/gallery-1.2.jpg"
  />
  <img
    className="gallery2"
    src="./img/home/gallery/gallery-2.1.jpg"
  />
  <img
    className="gallery3"
    src="./img/home/gallery/gallery-3.1.jpg"
  />
</section>
`;

exports[`HomeGallery snapshop - render 1`] = `
<section
  class="home-gallery"
>
  <div
    class="gallery-new-shoes"
  >
    <h2>
      NEW SHOES
    </h2>
    <p>
      We just dropped new fall shoe styles, and they are all ON SALE! Shop new combat boots, booties, mules, sneakers & more!
    </p>
    <button>
      SHOP SHOES
    </button>
  </div>
  <div
    class="gallery-accessories"
  >
    <h2>
      Accessories
    </h2>
    <p>
      Take your fall outfit to the next level with new accessories! The perfect trendy handbag, dainty gold necklace, coin bracelet or statement earrings will upgrade any look.
    </p>
    <button>
      SHOP ACCESSORIES
    </button>
  </div>
  <img
    class="gallery1"
    src="./img/home/gallery/gallery-1.2.jpg"
  />
  <img
    class="gallery2"
    src="./img/home/gallery/gallery-2.1.jpg"
  />
  <img
    class="gallery3"
    src="./img/home/gallery/gallery-3.1.jpg"
  />
</section>
`;

exports[`HomeGallery snapshop - shallow 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <HomeGallery />
</Router>
`;
